.github/workflows/CI.yml

```
@@ -655,6 +655,12 @@ jobs:
             terraform init
             if terraform apply -auto-approve -lock=false $opts -var="testing_ami=${{ matrix.testing_ami }}" -var="aoc_version=${{ needs.e2etest-preparation.outputs.version }}" -var="testcase=../testcases/${{ matrix.testcase }}" ; then terraform destroy -auto-approve ; else terraform destroy -auto-approve && exit 1 ; fi
 
+      #This is here just in case workflow cancel
+      - name: Destroy resources
+        if: ${{ always() && steps.e2etest-ec2-1.outputs.cache-hit != 'true' }}
```

```suggestion
        if: ${{ steps.e2etest-ec2-1.outputs.cache-hit != 'true' }}
```

if `always()` does what it says on the tin and always returns true, is it needed here?

---

Yes because it forces run on cancel according to the docs https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions